lib/grammers-crypto/src/two_factor_auth.rs

```diff
@@ -64,12 +64,17 @@ pub fn calculate_2fa(
     let k_v = (big_k * big_v) % &big_p;
 
     // t := (g_b - k_v) mod p (positive modulo, if the result is negative increment by p)
+    let mut changed_sign = false;
     let sub = if big_g_b > k_v {
         big_g_b - k_v
     } else {
+        changed_sign = true;
         k_v - big_g_b
     };
-    let big_t = sub % &big_p;
+    let mut big_t = sub % &big_p;
+    if changed_sign {
+        big_t = (&big_p - big_t) % &big_p;
+    }
```

Comment:
We could avoid mut variables by doing the equivalent:

```suggestion
    let sub = big_g_b - k_v;
    let big_t = if sub < BigUint::from(0u32) {
        (sub % &big_p) + &big_p
    } else {
        sub % &big_p
    };
```